autoload/inline_edit.vim

```diff
@@ -172,3 +172,84 @@ function! inline_edit#HereDoc()
 
   return [start, end, filetype, indent]
 endfunction
+
+" function! inline_edit#PythonQuotedString() {{{2
+
+function s:check_inside_python_string()
+  return index(map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")'), "pythonString") >= 0
+endfunction
+" Opens up a new proxy buffer with the contents of a fenced code block in
+" github-flavoured markdown.
+function! inline_edit#PythonMultilineString()
+  call inline_edit#PushCursor()
+
+  normal! 0
+
+  if !s:check_inside_python_string()
+    call inline_edit#PopCursor()
+    return []
+  endif
```

Comment:
One way to avoid popping the cursor before every return is to do it in a try-finally block:

``` vim
call inline_edit#PushCursor()

try
  "do stuff
  return []
finally
  call inline_edit#PopCursor()
endtry
```

In this case, you can wrap the entire contents of the function in the try-finally and pop the cursor at the end, since you only use it to maintain the same position, and don't modify it.

Reply:
That's cool! Never thought of using finally blocks like that in VimScript